cmake/Platform/Mac/PreInstallSteps_mac.cmake.in

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+cmake_minimum_required(VERSION 3.20)
+
+# The O3DE SDK will be shipped as an app bundle. So we create an O3DE_SDK.app directory
+# and install SDK into the app's Contents/Engine directory.
+set(LY_INSTALL_PATH_ORIGINAL ${CMAKE_INSTALL_PREFIX})
+file(MAKE_DIRECTORY ${CMAKE_INSTALL_PREFIX}/O3DE_SDK.app/Contents/MacOS)
+file(MAKE_DIRECTORY ${CMAKE_INSTALL_PREFIX}/O3DE_SDK.app/Contents/MacOS/Registry)
+
+file(COPY @LY_ROOT_FOLDER@/cmake/Platform/Mac/info.plist
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/O3DE_SDK.app/Contents
+)
```

Comment:
This could use the `ly_install_files` command to install the info.plist to the correct location within the SDK layout.
```suggestion
ly_install_files(FILES @LY_ROOT_FOLDER@/cmake/Platform/Mac/info.plist DESTINATION O3DE_SDK.app/Contents)
```

Also FYI the `LY_INSTALL_PATH_ORIGINAL` variable is unused

Reply:
Ok so it looks like this code is trying to update the `CMAKE_INSTALL_PREFIX` after installing some files to the "original" INSTALL_PREFIX root. 

In this case the [file(INSTALL)](https://cmake.org/cmake/help/v3.21/command/file.html?highlight=file#install) command should be able to create the intermediate directories as well

Reply:
LY_INSTALL_PATH_ORIGINAL is used in InstallUtils_mac.cmake to codesign the outermost(O3DE_SDK.app) bundle.

Reply:
I'll update to use file(INSTALL)

Reply:
Fixed.